### PR TITLE
DT-1893: 🐛 Fix info endpoint not working after java upgrade

### PR DESCRIPTION
**Pull request by petergphillips**

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -68,4 +69,8 @@ tasks {
       jvmTarget = "15"
     }
   }
+
+  test {
+    jvmArgs("--add-opens", "java.base/java.lang=ALL-UNNAMED")
```
Comment (petergphillips) [inline build.gradle.kts:74]: this is for running the tests, had to put the same change in the Dockerfile for running the app

File: src/main/resources/application.properties [hunk 2]
```diff
@@ -32,7 +32,5 @@ api.keyworker.deallocate.backoffMs=5000
 application.insights.ikey=
 
 logging.uris.exclude.regex=.*/health|.*/health/ping|.*/info|.*/ping|.*/swagger.*|/webjars/.*|.*/v2/api-docs
-graceful.shutdown.enabled=true
```
Comment (petergphillips) [inline src/main/resources/application.properties:35]: this was old way of shutting down

File: src/main/resources/application.yml [hunk 3]
```diff
@@ -48,6 +48,11 @@ spring:
   codec:
     max-in-memory-size: 10MB
 
+  profiles:
+    group:
+      test:
+        - "dev"
```
Comment (petergphillips) [inline src/main/resources/application.yml:54]: not having :point_up: meant that we didn't get any logging when running the tests

File: src/test/java/uk/gov/justice/digital/hmpps/keyworker/integration/AutoAllocationIntegrationTest.kt [hunk 4]
```diff
@@ -23,7 +23,7 @@ class AutoAllocationIntegrationTest : IntegrationTest() {
   @Test
   fun beforeEach() {
     migratedFoAutoAllocation(PRISON_ID)
-    eliteMockServer.stubOffendersAtLocationForAutoAllocation(PRISON_ID, OFFENDERS_AT_LOCATION)
```
Comment (petergphillips) [inline src/test/java/uk/gov/justice/digital/hmpps/keyworker/integration/AutoAllocationIntegrationTest.kt:26]: unused parameter

File: src/test/java/uk/gov/justice/digital/hmpps/keyworker/integration/AutoAllocationIntegrationTest.kt [hunk 5]
```diff
@@ -141,13 +141,4 @@ class AutoAllocationIntegrationTest : IntegrationTest() {
       .exchange()
       .expectStatus().is2xxSuccessful
   }
-
-  fun setKeyworkerCapacity(prisonId: String, keyworkerId: Long, capacity: Int) {
```
Comment (petergphillips) [inline src/test/java/uk/gov/justice/digital/hmpps/keyworker/integration/AutoAllocationIntegrationTest.kt:145]: moved to parent

File: src/test/java/uk/gov/justice/digital/hmpps/keyworker/integration/KeyworkerServiceIntegrationTest.kt [hunk 6]
```diff
@@ -20,10 +20,26 @@ class KeyworkerServiceIntegrationTest : IntegrationTest() {
   @Test
   fun beforeEach() {
     migratedFoAutoAllocation(PRISON_ID)
-    eliteMockServer.stubOffendersAtLocationForAutoAllocation(PRISON_ID, OFFENDERS_AT_LOCATION)
+    eliteMockServer.stubOffendersAtLocationForAutoAllocation(OFFENDERS_AT_LOCATION)
     eliteMockServer.stubKeyworkerRoles(PRISON_ID, KEYWORKER_ID_1, STAFF_LOCATION_ROLE_LIST)
   }
 
+  @Test
+  fun `Allocation history for offender reports ok`() {
```
Comment (petergphillips) [inline src/test/java/uk/gov/justice/digital/hmpps/keyworker/integration/KeyworkerServiceIntegrationTest.kt:28]: found out that was no test for the allocation history for an offender endpoint, needed one in order to load ehcache with some data

File: src/test/java/uk/gov/justice/digital/hmpps/keyworker/integration/health/InfoIntTest.kt [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+      getWiremockResponse("prisoners_information_A1234AA")
+    )
+
+    webTestClient.get()
+      .uri("/key-worker/allocation-history/$NON_MIGRATED_ALLOCATION_OFFENDER_ID")
+      .headers(setOmicAdminHeaders())
+      .exchange()
+      .expectStatus().is2xxSuccessful
+
+    webTestClient.get().uri("/info")
```
Comment (petergphillips) [inline src/test/java/uk/gov/justice/digital/hmpps/keyworker/integration/health/InfoIntTest.kt:61]: annoyingly the `/info` endpoint works fine when there isn't anything in the cache so need :point_up: to populate it first

File: src/test/java/uk/gov/justice/digital/hmpps/keyworker/integration/wiremock/EliteMockServer.kt [hunk 8]
```diff
@@ -67,7 +67,7 @@ class EliteMockServer : WireMockServer(9999) {
     )
   }
 
-  fun stubOffendersAtLocationForAutoAllocation(prisonId: String, json: String) {
```
Comment (petergphillips) [inline src/test/java/uk/gov/justice/digital/hmpps/keyworker/integration/wiremock/EliteMockServer.kt:70]: unused

File: src/test/resources/application-test.yml [hunk 9]
```diff
@@ -1,6 +1,3 @@
-server:
-  shutdown: immediate
```
Comment (petergphillips) [inline src/test/resources/application-test.yml:2]: moved to the application-dev.yml

**Review by Mjwillis — APPROVED** [review]
lgtm

**Pull request merged**